src/main/java/BossMan.java

```diff
@@ -0,0 +1,147 @@
+import java.util.Scanner;
+
+public class BossMan {
+    private static final String SEP = "____________________________________________________________";
+    private final Scanner SCANNER;
+    private final TaskList TASK_LIST;
+
+    public BossMan() {
+        this.SCANNER = new Scanner(System.in);
+        this.TASK_LIST = new TaskList();
+    }
+
+    public void greetUser() {
+        System.out.println(SEP + "\nHello! I'm BossMan");
+        System.out.println("What can I do for you?\n" + SEP);
+    }
+
+    public void endChat() {
+        System.out.println("Bye. Hope to see you again soon!\n" + SEP);
+    }
+
+    public void startChat() {
+        String userInput;
+
+        do {
+            System.out.print("You: ");
+            userInput = SCANNER.nextLine();
+
+            processUserInput(userInput);
+
+        } while (!userInput.equalsIgnoreCase("bye"));
+    }
+
+    private void processUserInput(String userInput) {
+        String[] parts = parseUserInput(userInput);
+
+        if (parts.length == 0) {
+            System.out.println("Unknown command\n" + SEP);
+            return;
+        }
+
+        String commandType = parts[0];
+        String commandArgs = parts.length > 1 ? parts[1] : "";
+
+        switch (commandType.toLowerCase()) {
+        case "mark":
+            handleMarkCommand(commandArgs);
+            break;
+
+        case "unmark":
+            handleUnmarkCommand(commandArgs);
+            break;
+
+        case "list":
+            TASK_LIST.printTasks();
+            break;
+
+        case "todo":
+            handleTodoCommand(commandArgs);
+            break;
+
+        case "deadline":
+            handleDeadlineCommand(commandArgs);
+            break;
+
+        case "event":
+            handleEventCommand(commandArgs);
+            break;
+
+        case "bye":
+            break; // No need to print unknown command for 'bye'
+
+        default:
+            System.out.println("Unknown command\n" + SEP);
+        }
+    }
+
+    private String[] parseUserInput(String userInput) {
+        return userInput.trim().split("\\s+", 2);
+    }
+
+    private void handleMarkCommand(String commandArgs) {
```

Comment:
This seems to be handled similarly. Consider refactoring it into a separate method to avoid duplication